tests/unit/test_client.py

```diff
@@ -1002,3 +1002,95 @@ def __call__(self):
     with_retry(FailerUntil(2).__call__)()
     with pytest.raises(SomeException):
         with_retry(FailerUntil(3).__call__)()
+
+
+def assert_headers_with_role(headers, role):
+    catalog = "test_catalog"
+    schema = "test_schema"
+    user = "test_user"
+    source = "test_source"
+
+    assert headers[constants.HEADER_CATALOG] == catalog
+    assert headers[constants.HEADER_SCHEMA] == schema
+    assert headers[constants.HEADER_SOURCE] == source
+    assert headers[constants.HEADER_USER] == user
+    assert headers[constants.HEADER_SESSION] == ""
+    assert headers[constants.HEADER_ROLE] == role
+    assert len(headers.keys()) == 7
+
+
+def test_request_headers_role_hive_all(mock_get_and_post):
+    get, post = mock_get_and_post
+    req = TrinoRequest(
+        host="coordinator",
+        port=8080,
+        client_session=ClientSession(
+            user="test_user",
+            source="test_source",
+            catalog="test_catalog",
+            schema="test_schema",
```

Comment:
Remove `catalog`, `schema` and `source` as it unrelates to role tests. Please fix `test_request_headers_role_admin` & `test_request_headers_role_empty` as well.

Reply:
:+1: